src/userobjects/OpenMCDomainFilterEditor.C

```diff
@@ -0,0 +1,201 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCDomainFilterEditor.h"
+#include "OpenMCProblemBase.h"
+#include "openmc/tallies/tally.h"
+#include "openmc/tallies/filter.h"
+#include "openmc/tallies/filter_cell.h"
+#include "openmc/tallies/filter_material.h"
+#include "openmc/tallies/filter_universe.h"
+#include "openmc/tallies/filter_mesh.h"
+
+registerMooseObject("CardinalApp", OpenMCDomainFilterEditor);
+
+InputParameters
+OpenMCDomainFilterEditor::validParams()
+{
+  InputParameters params = OpenMCUserObject::validParams();
+  params.addParam<bool>("create_filter", false, "Whether to create the filter if it doesn't exist");
+  params.addRequiredParam<int32_t>("filter_id", "The ID of the filter to modify");
+  params.addRequiredParam<MooseEnum>("filter_type", getFilterTypeEnum(), "The type of filter to create");
+  params.addRequiredParam<std::vector<std::string>>("bins", "The bins to modify in the filter");
+  params.declareControllable("bins");
+  params.addClassDescription("A UserObject for creating and managing OpenMC domain tally filters");
+  return params;
+}
+
+OpenMCDomainFilterEditor::OpenMCDomainFilterEditor(const InputParameters & parameters)
+  : OpenMCUserObject(parameters),
+    _create_filter(getParam<bool>("create_filter")),
+    _filter_id(getParam<int32_t>("filter_id")),
+    _filter_type(getParam<MooseEnum>("filter_type").getEnum<OpenMCFilterType>())
+{
+  this->initialize();
```

Comment:
Same question here about `initialize()` -- does this need to be here, or is it sufficient for MOOSE to call `initialize()` during the typical setup patterns?